src/02-room-details.js

```diff
@@ -49,9 +63,36 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+function getConnectedRoomNamesById(rooms, id) {
+  // Variable that will contain the names of the rooms that are connected
+  let connectedArr = [];
+  // Variable that will contain every rooms object's roomId
+  let roomIdArr = [];
+ 
+  // Loop through the rooms array and push every roomId into the roomIdArr variable
+  for (let room of rooms) {
+    roomIdArr.push(room.roomId);
+    // Push the names of the rooms that contain the parameter id in the room.connectsTo array into the arr array
+    if (room.connectsTo.includes(id)) {
+      connectedArr.push(room.name);
+    }
+  }
+  // Loop through each room object and then loop through each object's connectsTo array and return an error message if every element in the connects to array exists in the roomIdArr array
+  for (let room of rooms) {
+    for (let ele of room.connectsTo){
+      if (!roomIdArr.includes(ele)) {
+        return `Room with ID of 'incorrect-id' could not be found.`;
```

Comment:
this is considered hardcoded. What if the id was `abc1234`, we would still want to get this error message with the incorrect id dynamically inserted in the string. 

In order to check which ID is incorrect you would have to look for each ID in the `rooms` array and the one that is not found you would return the error message